cdc/sink/dmlsink/cloudstorage/dml_worker.go

```diff
@@ -237,10 +237,26 @@ func (d *dmlWorker) writeDataFile(ctx context.Context, path string, task *single
 		callbacks = append(callbacks, msg.Callback)
 	}
 
-	if err := d.statistics.RecordBatchExecution(func() (int, error) {
-		err := d.storage.WriteFile(ctx, path, buf.Bytes())
-		if err != nil {
-			return 0, err
+	if err := d.statistics.RecordBatchExecution(func() (_ int, inErr error) {
+		if d.config.FlushConcurrency <= 1 {
+			return rowsCnt, d.storage.WriteFile(ctx, path, buf.Bytes())
+		}
+
+		writer, inErr := d.storage.Create(ctx, path, &storage.WriterOption{
```

Comment:
This is just use the multipart upload feature of S3 under the hood? https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html

Reply:
Yes, this optimization is currently only available for s3.